src/detector/implementations/picamera2_implementation.py

```diff
@@ -0,0 +1,181 @@
+from src.detector.api import \
+    GetCaptureDeviceResponse, \
+    GetCaptureImageRequest, \
+    GetCaptureImageResponse, \
+    GetCapturePropertiesResponse, \
+    SetCaptureDeviceRequest, \
+    SetCapturePropertiesRequest
+from src.common import \
+    EmptyResponse, \
+    ErrorResponse, \
+    get_kwarg, \
+    MCastResponse
+from src.common.structures.capture_status import CaptureStatus
+
+from src.detector.implementations import AbstractCameraInterface
+
+from picamera2 import Picamera2, Preview
+
+import base64
+import cv2
+import datetime
+import logging
+import time
+import numpy
+import os
+from typing import Any, Callable
+
+logger = logging.getLogger(__name__)
+
+class PiCamera(AbstractCameraInterface):
+
+    _capture: numpy.ndarray | None
+    _captured_image: numpy.ndarray | None
+
+    def __init__(self):
+        self._capture = None
+        self._captured_image = None
+        self._captured_timestamp_utc = datetime.datetime.min
+
+        self._capture_status = CaptureStatus
+        self._capture_status.status = CaptureStatus.Status.STOPPED
+
+        self._captured_timestamp_utc = datetime.datetime.min
+
+        self._capture_status = CaptureStatus
+        self._capture_status.status = CaptureStatus.Status.STOPPED
+
+        # TODO: DEBUGGING
+        self._capture_status.status = CaptureStatus.Status.RUNNING
+
+        self._camera = Picamera2()
+        self._camera.start()
+
+    def __del__(self):
+        if self._capture is not None:
+            self._capture = numpy.empty
+
+    def _detect_os_and_open_video(self):
+        return self._camera.capture_array()
+
+    def internal_update_capture(self):
+        self._captured_image = self._camera.capture_array()
+
+        if self._captured_image is None:
+            message: str = "Failed to grab frame."
+            self._status.capture_errors.append(message)
+            self._capture_status.status = CaptureStatus.Status.FAILURE
+            return ("error", message)
```

Comment:
Okay, I see. So this returns a `tuple` of str, str, not just a `str`. I suggest to add type hint to the function declaration, and that you make sure to unpack the variables properly when calling this function